tests/README.md

```diff
@@ -0,0 +1,1510 @@
+
+| Europe/Warsaw time  |   offset |  UTC                |    php2utc conversion |  php offset  |
+| ------------------- | -------- | ------------------- | --------------------- | ------------ |
+| 2013-10-27 01:00:00 |    +2    | 2013-10-26 23:00:00 |   2013-10-26 23:00:00 | +2           |
+| 2013-10-27 01:30:00 |    +2    | 2013-10-26 23:30:00 |   2013-10-26 23:30:00 | +2           |
+| 2013-10-27 02:00:00 |    +2    | 2013-10-27 00:00:00 |   2013-10-27 01:00:00 | **+1**       |
+| 2013-10-27 02:30:00 |    +2    | 2013-10-27 00:30:00 |   2013-10-27 01:30:00 | **+1**       |
+| 2013-10-27 02:59:00 |    +2    | 2013-10-27 00:59:00 |   2013-10-27 01:59:00 | **+1**       |
+
+3am -> 2am .....................................summer time changes to standard(winter) time @3am we subtract 1h so 3am becomes 2am
+
+| Europe/Warsaw time  |   offset |  UTC                  |    php2utc conversion  |  php offset  |
+| ------------------- | -------- | --------------------- | ---------------------- | ------------ |
+| 2013-10-27 02:00:00 |    +1    | 2013-10-27 01:00:00   | 2013-10-27 01:00:00    | +1           |
+| 2013-10-27 02:30:00 |    +1    | 2013-10-27 01:30:00   | 2013-10-27 01:30:00    | +1           |
+| 2013-10-27 03:00:00 |    +1    | 2013-10-27 02:00:00   | 2013-10-27 02:00:00    | +1           |
+| 2013-10-27 03:30:00 |    +1    | 2013-10-27 02:30:00   | 2013-10-27 02:30:00    | +1           |
+
+
+<a name="test-examples"></a>
+## Test Examples
+This test section will show good and bad unit tests in ILIAS.
+
+<a name="negative-examples"></a>
+### Negative Examples
+All the examples are shown in this section have some different problems which will be explained in detail. However
+it is really important to understand that only the code is wrong. The statements made are never about a developers skills and how they are now
+in the present. "It's just code!"
+
+<a name="wrong-location-and-bloated"></a>
+#### Wrong Location and Bloated
+The test shown below is the only test of the course module. However the code actually test the
+*ilMemberAgreement* class which is part of the Membership service.
+
+In addition this test tests more than one thing:
+- Agreement of a user to some policy
+- Removal of a user from the acceptance list of a policy
+- Test of a second static method which makes the same
+- Fetch the agreement with the object id 8888.
+- Removal of a user with the user id 9999, however the result is not evaluated.
+
+The test requires also an initialised ILIAS instance because the values are written to the db which could lead
+to an unpredictable result if an other test maybe an other *ilMemberAgreement* test class manipulates the database.
+
+Finally in this case it should be considered to remove the test entirely because it is not of real use. In addition this test is
+skipped on the CI server which leads to the final question if test is ever executed.
```

Comment:
That could be true indeed, but there is no definition of integration tests in ILIAS so far. I think it would be nice to have something like that in the future, and remove this example at that point.